svn/common.py

```diff
@@ -450,12 +453,18 @@ def diff(self, old, new, rel_path=None):
             do_combine=True)
         file_to_diff = {}
         for non_empty_diff in filter(None, diff_result.decode('utf8').split('Index: ')):
-            split_diff = non_empty_diff.split('==')
+            split_diff = \
```

Comment:
Why does this need to change?

Reply:
The current implementation would split the overall diff text any time it finds a ==, including in places like ``if (a == b)``. Without an equivalent change, the diff text will be partially lost.

I can think of prettier ways to do it, say something like ``'='*67`` instead of having a big long list. 

I provided the unit test ``test_diff_separator`` for this one, which checks for code above and below a double-equals in a known file. It passes with the new implementation, but fails with the old one.